### PR TITLE
optimize command simulator auto-generation

### PR DESCRIPTION
**Pull request by chris-rock**
- run `bundle exec rake update_demo` from the root, and everything is magic then
- all commands are simulated in a docker container now
- builds the simulator container with InSpec and the mock backend
- moves all generated content to `app/responses`
- merges `commands.json` and `extra_commands.json`
- move rake scripts to `www/turorial/scripts`
- simulate README.md via simulator now
- updates examples
- updates README.md

### CODE REVIEW

**Comment by vjeffrey** [comment]
got this error running the rake task: 
screen shot 2016-09-15 at 9 06 39 pm
changing line 50 of   build_simulator_runtime.rb to FileUtils.cp_r(Dir.glob(pwd+'/examples/profile/*'), pwd + '/simulator/filesystem/examples/profile') fixed it

**Comment by vjeffrey** [comment]
holy awesome awesomeness :)  this is so cool chris.  i would love it if you could walk me through how you did all the updates sometime whenever you have time  :)

File: www/tutorial/README.md [hunk 1]
```diff
@@ -1,61 +1,45 @@
...
 Webapp then went and found her friend xterm, and said, why let's do this together, for two heads are always better than one!
 And so webapp said to her dear friend angular2, please help me make this work! and angular2 came running, as friends always do.
 And so angular2 met xterm, and as they shook hands, they knew they would be good friends together. So they put their heads together,
-and made some decisions about what each one's role would be. Xterm claimed responsiblity for creating the terminal view,
+and made some decisions about what each one's role would be. Xterm claimed responsibility for creating the terminal view,
 reading the user's input, and displaying it.  App then explained she could take that final user input sent over by xterm and
-and transform it into a regular expression, which she could then use to match against the information in the commandsjson file,
+and transform it into a regular expression, which she could then use to match against the information in the `commands.json` file,
 matching the user's input against the key and then retrieving the associated value's txt file.
 And it was so. :)
```
Comment (vjeffrey) [inline www/tutorial/README.md:36]: We should probably remove my silly paragraph above here and just explain how it works normally, now that's it's so awesome :)  no one wants to read a whole paragraph thing; that was just the result of wine + code

**Comment by chris-rock** [comment]
@vjeffrey Thanks for the feedback, I added an empty profile directory to this PR. Therefore the script should now work trough.

File: www/tutorial/README.md (hunk 1 above)
Comment (chris-rock) [reply www/tutorial/README.md:36]: I was not sure, I liked the idea somehow, but we could shorten that up a bit :-)

**Comment by vjeffrey** [comment]
:+1: works wonderfully :)

**Pull request merged**